Chessington.GameEngine/Pieces/Bishop.cs

```diff
@@ -10,7 +10,76 @@ public Bishop(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+
+            for (int i = board.FindPiece(this).Row + 1, j = board.FindPiece(this).Col + 1; i < 8 && j < 8; i++, j++)
+            {
+                if (board.GetPiece(new Square(i, j)) == null)
+                {
+                    availableMoves.Add(new Square(i, j));   
+                }
+                else if (board.GetPiece(new Square(i, j)).Player != this.Player)
+                {
+                    availableMoves.Add(new Square(i, j));
+                    break;
+                }
+                else
+                {
+                    break;
+                }
+            }
+            
```

Comment:
If you implement a GoInDirection(Square startingPosition, int colDir, int rowDir) function (the same from the Rook ) you could shorten this code again eg. If you want the bishop to go in the direction up-right rowDir = -1 and colDir = 1.